index/scorch/scorch.go

```diff
@@ -515,9 +515,26 @@ func (s *Scorch) diskFileStats(rootSegmentPaths map[string]struct{}) (uint64,
 	return numFilesOnDisk, numBytesUsedDisk, numBytesOnDiskByRoot
 }
 
-func (s *Scorch) rootDiskSegmentsPaths() map[string]struct{} {
-	rv := make(map[string]struct{}, len(s.root.segment))
-	for _, segmentSnapshot := range s.root.segment {
+// reClaimableDocsRatio gives a ratio about the obsoleted or
+// reclaimable documents present in a given index snapshot.
+func reClaimableDocsRatio(indexSnapshot *IndexSnapshot) float64 {
+	var totalCount, liveCount uint64
+	for _, segmentSnapshot := range indexSnapshot.segment {
+		if _, ok := segmentSnapshot.segment.(segment.PersistedSegment); ok {
+			totalCount += uint64(segmentSnapshot.FullSize())
+			liveCount += uint64(segmentSnapshot.Count())
+		}
+	}
+
+	if totalCount > 0 {
+		return float64(totalCount-liveCount) / float64(totalCount)
+	}
+	return 0
+}
+
+func rootDiskSegmentsPaths(indexSnapshot *IndexSnapshot) map[string]struct{} {
```

Comment:
So, I think this refactoring is good, but not quite complete.

Now that this function works on `*IndexSnapshot`, I think it should be method on that type.  Second, the function now misnamed, as it returns the disk segment paths for whichever snapshot we pass in.  We will just use it to find the disk segment paths of the root, but it works for any snapshot.

So, together this would give us:

```func (i *IndexSnapshot) diskSegmentPaths() map[string]struct{}`

Reply:
absolutely make sense.
 Shall we make reClaimableDocsRatio also of IndexSnapshot  type?

Reply:
Yeah I think that makes sense too.